chainer/gradient_check.py

```diff
@@ -114,3 +115,131 @@ def assert_allclose(x, y, atol=1e-5, rtol=1e-4, verbose=True):
     except Exception:
         print('error:', numpy.abs(x - y).max())
         raise
+
+
+def _as_tuple(x):
+    if isinstance(x, tuple):
+        return x
+    elif isinstance(x, list):
+        return tuple(x)
+    else:
+        return (x,)
+
+
+def check_backward(func, x_data, y_grad, params=(),
+                   eps=1e-3, atol=1e-5, rtol=1e-4):
+    """Test backward procedure of a given function.
+
+    This function automatically check backward-process of a given function.
+    For example, when you have a :class:`~chainer.Function` class ``MyFunc``,
+    that gets two arguments and returns one value, you can make its test like
+    this::
+
+    >> def test_my_func(self):
+    >>   func = MyFunc()
+    >>   x1_data = xp.array(...)
+    >>   x2_data = xp.array(...)
+    >>   gy_data = xp.array(...)
+    >>   check_backward(func, (x1_data, x2_data), gy_data)
+
+    This method creates :class:`~chainer.Variable` objects with ``x_data``,
+    and call ``func`` with the :class:`~chainer.Variable`s to get its result
+    :class:`~chainer.Variable`.
+    And then, set ``grad`` of the results with ``y_grad`` arrays and call
+    ``backward`` method to get gradients of the inputs.
+    To check correctness of the gradients, the function calls
+    :func:`numerical_grad` and compares them with :func:`assert_allclose`.
+    If an input object represents integer variable, its gradient is ignored.
+
+    You can simplify a test when ``MyFunc`` gets only one argument::
+
+    >>   check_backward(func, x1_data, gy_data)
+
+    If ``MyFunc`` is a loss function which returns zero-dimensional
+    array, pass ``None`` to ``gy_data``::
+
+    >>   check_backward(my_loss_func, (x1_data, x2_data), None)
```

Comment:
Please write this functions's behavior in this case (set 1 to ``grad`` attribute of the results).